From 13ae8418f93b9213f46122e3a3e55bec89fefa2c Mon Sep 17 00:00:00 2001
From: Aishwarya R <quic_aisr@quicinc.com>
Date: Wed, 6 Apr 2022 13:12:17 +0530
Subject: [PATCH] ath12k: save max tx power in vdev start response event from
 firmware

Save the max tx power received in the vdev start response event from
firmware. A subsequent patch will use this to calculate the final power
value for WMI_VDEV_SET_TPC_POWER_CMDID.

Signed-off-by: Aishwarya R <quic_aisr@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/core.h | 1 +
 drivers/net/wireless/ath/ath12k/wmi.c  | 2 ++
 drivers/net/wireless/ath/ath12k/wmi.h  | 1 +
 3 files changed, 4 insertions(+)

--- a/drivers/net/wireless/ath/ath12k/core.h
+++ b/drivers/net/wireless/ath/ath12k/core.h
@@ -662,6 +662,7 @@ struct ath12k {
 	bool monitor_started;
 	int monitor_vdev_id;
 	u8 twt_enabled;
+	s8 max_allowed_tx_power;
 };
 
 struct ath12k_band_cap {
--- a/drivers/net/wireless/ath/ath12k/wmi.c
+++ b/drivers/net/wireless/ath/ath12k/wmi.c
@@ -4892,6 +4892,7 @@ static int ath12k_pull_vdev_start_resp_t
 	vdev_rsp->mac_id = ev->mac_id;
 	vdev_rsp->cfgd_tx_streams = ev->cfgd_tx_streams;
 	vdev_rsp->cfgd_rx_streams = ev->cfgd_rx_streams;
+	vdev_rsp->max_allowed_tx_power = ev->max_allowed_tx_power;
 
 	kfree(tb);
 	return 0;
@@ -6918,6 +6919,7 @@ static void ath12k_vdev_start_resp_event
 	}
 
 	ar->last_wmi_vdev_start_status = 0;
+	ar->max_allowed_tx_power = vdev_start_resp.max_allowed_tx_power;
 
 	status = vdev_start_resp.status;
 
--- a/drivers/net/wireless/ath/ath12k/wmi.h
+++ b/drivers/net/wireless/ath/ath12k/wmi.h
@@ -4222,6 +4222,7 @@ struct wmi_vdev_start_resp_event {
 	};
 	u32 cfgd_tx_streams;
 	u32 cfgd_rx_streams;
+	s32 max_allowed_tx_power;
 } __packed;
 
 /* VDEV start response status codes */
